synthesizer/tests/utilities/traits.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+/// A general interface for a test.
+pub trait Test: Sized {
+    /// Loads the test from the given path.
+    fn load<P: AsRef<Path>>(input: P) -> Result<Self>;
```

Comment:
Env vars are sometimes hard to work with. This seems maybe be the beginning of a linting system, so a potential idea to make it more flexible is to allow a config object to be specified in the method directly.

```suggestion
    fn load<P: AsRef<Path>, T>(input: P, config: Option<&T>) -> Result<Self>;
```

Reply:
That's a fair point, however adding a generic `config` is challenging to work into the test `Runner`.
The intention was that an implementation of `Test` or `Expectation` specifies how to extract its configuration.
Usually, this would be from reading the test or expectation file.

That said, it would be good to revisit this idea if we have a need to extend the test framework.
Also for context, I hadn't considered how this would function as a linting framework.

Reply:
Yeah agreed, we don't want to make it took complicated before it needs to be. Just a thought for the future :)